app/controllers/catalog_controller.rb

```diff
@@ -175,10 +177,12 @@ def atomic_form_field_changed
       @record = class_service_template(params[:st_prov_type]).new
       set_form_vars
       @edit[:new][:st_prov_type] = params[:st_prov_type] if params[:st_prov_type]
+      @edit[:new][:generic_subtype] = params[:generic_subtype] if params[:generic_subtype]
       @edit[:new][:service_type] = "atomic"
       default_entry_point(@edit[:new][:st_prov_type]) if params[:st_prov_type].start_with?('generic')
       @edit[:rec_id] = @record ? @record.id : nil
       @tabactive = @edit[:new][:current_tab_key]
+      @generic_subtypes = generic_item_subtypes
```

Comment:
@ZitaNemeckova save it in ``` @edit ``` so you don't have to set it every time form_field change transaction comes up.